src/main/java/htmlpublisher/HtmlPublisher.java

```diff
@@ -195,30 +196,36 @@ public static boolean publishReports(Run<?, ?> build, FilePath workspace, Launch
             e1.printStackTrace();
             return false;
         }
-        
+
         for (int i=0; i < reportTargets.size(); i++) {
             // Create an array of lines we will eventually write out, initially the header.
             ArrayList<String> reportLines = new ArrayList<String>(headerLines);
-            HtmlPublisherTarget reportTarget = reportTargets.get(i); 
+            HtmlPublisherTarget reportTarget = reportTargets.get(i);
             boolean keepAll = reportTarget.getKeepAll();
             boolean allowMissing = reportTarget.getAllowMissing();
-            
+
             FilePath archiveDir = workspace.child(resolveParametersInString(build, listener, reportTarget.getReportDir()));
             FilePath targetDir = reportTarget.getArchiveTarget(build);
-            
-            String levelString = keepAll ? "BUILD" : "PROJECT"; 
+
+            String levelString = keepAll ? "BUILD" : "PROJECT";
             listener.getLogger().println("[htmlpublisher] Archiving at " + levelString + " level " + archiveDir + " to " + targetDir);
 
             // The index name might be a comma separated list of names, so let's figure out all the pages we should index.
             String[] csvReports = resolveParametersInString(build, listener, reportTarget.getReportFiles()).split(",");
+
+            String[] titles = null;
+            if (reportTarget.getReportTitles() != null && reportTarget.getReportTitles().indexOf(",") > 0) {
```

Comment:
If you only have one report file, the name doesn't apply because of this logic. Is that intentional?